rcl_action/include/rcl_action/types.h

```diff
@@ -250,52 +190,40 @@ rcl_action_cancel_request_init(
  * Uses Atomics       | No
  * Lock-Free          | Yes
  *
- * \param[inout] cancel_request the cancel request message to be deinitialized
- * \param[in] action_server handle to the action sever used to create the cancel request
- * \return `RCL_RET_OK` if the cancel request was deinitialized successfully, or
+ * \param[inout] status_array the goal status array message to be deinitialized
+ * \param[in] allocator handle to the allocator used to create the goal status array
+ * \return `RCL_RET_OK` if the goal status array was deinitialized successfully, or
  * \return `RCL_RET_INVALID_ARGUMENT` if any arguments are invalid, or
- * \return `RCL_RET_ACTION_SERVER_INVALID` if the action server is invalid, or
  * \return `RCL_RET_ERROR` if an unspecified error occurs.
  */
 RCL_ACTION_PUBLIC
 RCL_WARN_UNUSED
 rcl_ret_t
-rcl_action_cancel_request_fini(
-  rcl_action_cancel_request_t * cancel_request,
-  const rcl_action_server_t * action_server);
-
-/// Return a rcl_action_cancel_response_t with members set to `NULL`.
-/**
- * Should be called to get a null rcl_action_cancel_response_t before passing to
- * rcl_action_cancel_response_init().
- */
-RCL_ACTION_PUBLIC
-RCL_WARN_UNUSED
-rcl_action_cancel_response_t
-rcl_action_get_zero_initialized_cancel_response(void);
+rcl_action_goal_status_array_fini(
+  rcl_action_goal_status_array_t * status_array,
+  const rcl_allocator_t allocator);
```

Comment:
Recommend storing allocator in `rcl_action_goal_status_array_t` to avoid accidentally deallocating with the wrong allocator.

Reply:
Currently, it is defined as:

```c
typedef action_msgs__msg__GoalStatusArray rcl_action_goal_status_array_t;
```

If we want to include space for the allocator, then I suppose we should do something like:

```c
typedef struct rcl_action_goal_status_array_t
{
  action_msgs__msg__GoalStatusArray msg;
  rcl_allocator_t allocator;
} rcl_action_goal_status_array_t;
```
I can make a follow-up PR to address this and your other comments.

Reply:
👍 came here to say this, I noticed it in the API when rebasing my pr (https://github.com/ros2/rcl/pull/314).